src/lnp/application/message.rs

```diff
@@ -591,6 +634,237 @@ pub struct ChannelReestablish {
     pub my_current_per_commitment_point: PublicKey,
 }
 
+/// Bolt 7 Gossip messages
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct AnnouncementSignatures {
+    /// The channel ID
+    pub channel_id: ChannelId,
+
+    /// Short channel Id
+    pub short_channel_id: ShortChannelId, //TODO
+
+    /// Node Signature
+    pub node_signature: Signature,
+
+    /// Bitcoin Signature
+    pub bitcoin_signature: Signature,
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct ChannelAnnouncements {
+    /// Node Signature 1
+    pub node_signature_1: Signature,
+
+    /// Node Signature 2
+    pub node_signature_2: Signature,
+
+    /// Bitcoin Signature 1
+    pub bitcoin_signature_1: Signature,
+
+    /// Bitcoin Signature 2
+    pub bitcoin_signature_2: Signature,
+
+    /// length of feature bytes
+    pub len: u16,
+
+    /// feature bytes
+    pub features: Vec<u8>,
+
+    /// chain hash
+    pub chain_hash: AssetId,
+
+    /// Short channel ID
+    pub short_channel_id: ShortChannelId,
+
+    /// Node Id 1
+    pub node_id_1: PublicKey,
+
+    /// Node Id 2
+    pub node_id_2: PublicKey,
+
+    /// Bitcoin key 1
+    pub bitcoin_key_1: PublicKey,
+
+    /// Bitcoin key 2
+    pub bitcoin_key_2: PublicKey,
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct NodeAnnouncements {
+    /// Signature
+    pub signature: Signature,
+
+    /// length of feature bytes
+    pub flen: u16,
+
+    /// feature bytes
+    pub features: Vec<u8>,
+
+    /// time stapm
+    pub timestamp: u32,
+
+    /// Node Id
+    pub node_id: PublicKey,
+
+    /// RGB colour code
+    pub rgb_color: NodeColor,
+
+    /// Node Alias
+    pub alias: Alias,
+
+    /// length of node address bytes
+    pub addrlen: u16,
+
+    /// Node address
+    pub address: Vec<u8>,
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct ChannelUpdate {
+    /// Signature
+    pub signature: Signature,
+
+    /// Chainhash
+    pub chain_hash: AssetId,
+
+    /// Short Channel Id
+    pub short_channel_id: ShortChannelId,
+
+    /// Time stamp
+    pub timestamp: u32,
+
+    /// message flags
+    pub message_flags: u8,
+
+    /// channel flags
+    pub channle_flags: u8,
```

Comment:
We need a big-flagged structures here. Need to investigate can we use native rust bit flag structure, or do a custom type. Feel free to make it an issue for the future work, we just need not to forget about it.

Reply:
Creating an issue for future work.